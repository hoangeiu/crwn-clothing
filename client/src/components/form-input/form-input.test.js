import React from "react";
import { shallow } from "enzyme";

import { FormInput } from "./form-input.component";

describe("FormInput component", () => {
  let wrapper;
  let mockHandleChange;
  let mockLabel;
  beforeEach(() => {
    mockHandleChange = jest.fn();
    mockLabel = "email";

    const mockProps = {
      label: mockLabel,
      value: "test@gmail.com",
      handleChange: mockHandleChange,
    };

    wrapper = shallow(<FormInput {...mockProps} />);
  });

  it("should FormInput render without error", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call handleChange when input changes", () => {
    wrapper.find(".form-input").simulate("change");

    expect(mockHandleChange).toHaveBeenCalled();
  });

  describe("check label", () => {
    it("should render label if there is a label", () => {
      expect(wrapper.exists(".form-input-label")).toBeTruthy();
      expect(wrapper.find(".form-input-label").text()).toBe(mockLabel);
    });

    it("should not render label if there is no label", () => {
      mockLabel = "";
      const mockProps = {
        label: mockLabel,
        value: "test@gmail.com",
        handleChange: mockHandleChange,
      };

      wrapper = shallow(<FormInput {...mockProps} />);

      expect(wrapper.exists(".form-input-label")).toBeFalsy();
    });
  });
});
