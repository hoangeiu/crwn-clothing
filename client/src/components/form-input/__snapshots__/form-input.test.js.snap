// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput component should FormInput render without error 1`] = `
<div
  className="group"
>
  <input
    className="form-input"
    onChange={[MockFunction]}
    value="test@gmail.com"
  />
  <label
    className="shrink form-input-label"
  >
    email
  </label>
</div>
`;
